packages/@azure/servicebus/data-plane/lib/clientEntityContext.ts

```diff
@@ -113,14 +118,23 @@ export namespace ClientEntityContext {
       entityPath: entityPath,
       requestResponseLockedMessages: new ConcurrentExpiringMap<string>(),
       isSessionEnabled: !!options.isSessionEnabled,
-      messageSessions: {}
+      messageSessions: {},
+      expiredMessageSessions: {}
     };
 
     (entityContext as ClientEntityContext).sessionManager = new SessionManager(
       entityContext as ClientEntityContext
     );
 
     (entityContext as ClientEntityContext).getReceiver = (name: string, sessionId?: string) => {
+      if (sessionId != undefined && entityContext.expiredMessageSessions[sessionId] === true) {
+        // TODO: Use an error constructor utility which could also serve as interface documentation for all types of errors thrown
+        const error = new Error();
+        error.name = "SessionLockLostError";
+        error.message = `Session Lock is lost for: ${sessionId}. Open a new Session using getSessionReceiver()`;
+        throw error;
+      }
```

Comment:
Now that we are throwing error from `getReceiver`, why not move the `Cannot find the receiver with name...` error in here as well and update the signature of `getReceiver` to not return `undefined`?

Reply:
I think we might still want to know if that ever happens, due to maybe some other type of AMQP error

Reply:
Yes, I am not saying we should include the `Cannot find the receiver with name...` inside the `if` block for session lock lost case.

I am saying we can include it just before returning the `receiver` from `getReceiver`.

If you look at the 4 places where we settle the message, all of them call `getReceiver` and then throw the `Cannot find the receiver with name...` error if the `receiver` is undefined. We can pull that in.